src/components/gallery/HdGalleryCarousel.vue

```diff
@@ -130,9 +130,11 @@ export default {
   },
   mounted() {
     this.$nextTick(() => {
-      this.updatePagerItemsCount();
-      this.$refs.flickity.on('change', this.updateCurrentIndex);
-      this.$refs.flickity.on('staticClick', this.onStaticClick);
+      if (this.$refs.flickity) {
```

Comment:
https://stackoverflow.com/a/3390426/9218664
<img width="666" alt="Screenshot 2022-04-11 at 12 32 24 PM" src="https://user-images.githubusercontent.com/14016028/162710201-b21fc09d-d7f1-4444-b636-8a367ead3aca.png">


Shouldn't we check the type here? something like:
```if (typeof this.$refs.flickity !== 'undefined')```

Reply:
In this case, we do not need to check the type of the object, we need to know if it is present at all. And there can be two options: either we check for the true or the false. I've chosen true checking and run the function only if the object is present. The following examples provides a complete list of JavaScript falsy values:
```
if (false)
if (null)
if (undefined)
if (0)
if (-0)
if (0n)
if (NaN)
if ("")
```
Therefore, the code will run the value is present, and will not be undefined. 😉